include/seqan3/range/container/concept.hpp

```diff
@@ -161,6 +161,8 @@ concept bool container_concept = requires (type val, type val2)
 
     { val.begin()     } -> typename type::iterator;
     { val.end()       } -> typename type::iterator;
+    { val.begin()     } -> typename type::const_iterator;
+    { val.end()       } -> typename type::const_iterator;
```

Comment:
you need to actually create const version of val, so that begin on it also returns const_iterator